src/library_fs.js

```diff
@@ -414,17 +414,23 @@ FS.staticInit();` +
           },
           isAppend: {
             get: function() { return (this.flags & {{{ cDefine('O_APPEND') }}}); }
-          }
+          },
+          get flags() { return this.shared.flags; },
+          set flags(value) { this.shared.flags = value; },
+          get position() { return this.shared.position; },
+          set position(value) { this.shared.position = value; },
         };
       }
       // clone it, so we can return an instance of FSStream
       stream = Object.assign(new FS.FSStream(), stream);
+      stream.shared.refcnt ++;
```

Comment:
So if I understand correctly this change means that instead of copying the flags/position of the stream we are cloning we copy the shared object which means that the two streams then share flags and position?   

Do the flags need to be shared?  Can they change over time?

Finally, it looks like the refcount thing is only used the rawfs, can we make this explicit by wrapping it of its usage in `#if NODERAWFS`?   Even better, is there some way to move all the refernce count handling into `library_noderawfs.js`?  Maybe not?

Reply:
I think maybe you can do this in `library_noderawfs.js`:

```
createStream: (stream, fd_start, fd_end) => {
   // Call the original FS.createStream
   var rtn = VFS.createStream(stream, fd_start, fd_end);
   if (typeof rtn.shared.refcnt == "undefined") {
     rtn.shared.refcnf = 0;
   } else {
     rtn.shared.refcnf += 1;
   }
   return rtn;
}
closeStream(fd) {
  FS.streams[fd].shared.refcnt --;
  VFS.closeStream(fd);
}
```

This would mean that all the recounting is handed in noderawfs without the need to complicate the normal filesystem.

Reply:
> means that the two streams then share flags and position?
>
> Do the flags need to be shared? Can they change over time?

In a real linux the flags can change and must be shared. I'm not sure if these behaviors are implemented in emscripten though.